softwareComponents/torqueComputation/src/linprog.cpp

```diff
@@ -0,0 +1,133 @@
+            throw std::logic_error("Objective function coefficients length should equal to number of columns of constraint matrix.");
+        }
+    }
+
+    void copyVecToArray(const vec& colVector, uword size, double * location) {
+        memcpy(location, colVector.begin(), size * sizeof(double));
+    }
+
+    OptimizeResult linprog(
+            const mat& constraintMatrix,
+            const vec& constraintLowerBounds,
+            const vec& constraintUpperBounds,
+            const vec& lowerBounds,
+            const vec& upperBounds,
+            const std::optional<vec>& objectiveCoefficients /*= std::nullopt*/
+    ) {
+        argsSanityCheck(
+                constraintMatrix,
+                constraintLowerBounds,
+                constraintUpperBounds,
+                lowerBounds,
+                upperBounds,
+                objectiveCoefficients
+        );
+
+        // Empty model
+        ClpSimplex  model;
+
+        // CLP support only ints for size of matrix.
+        int numberRows = constraintMatrix.n_rows & INT_MAX;;
+        int numberColumns = constraintMatrix.n_cols & INT_MAX;
+        int numberElements = numberRows * numberColumns & INT_MAX;
+        // Arrays will be set to default values
+        model.resize(numberRows, numberColumns);
+
+        double * elements = new double [numberElements];
+
+        CoinBigIndex * starts = new CoinBigIndex [numberColumns+1];
+
+        int * rows = new int [numberElements];
+        std::iota(rows, rows + numberRows, 0);
+        for (int i = 1; i < numberColumns; i++) {
+            memcpy(rows + i * numberRows, rows, numberRows * sizeof(int));
+        }
+
+        int * lengths = new int[numberColumns];
+        std::fill_n(lengths, numberColumns, numberRows);
+
+        double * columnLower = model.columnLower();
+        double * columnUpper = model.columnUpper();
+        double * objective = model.objective();
+        double * rowLower = model.rowLower();
+        double * rowUpper = model.rowUpper();
+
+        //Set objective
+        if (objectiveCoefficients != std::nullopt) {
+            copyVecToArray(objectiveCoefficients.value(), numberColumns, objective);
+        }
+
+        // Set x bounds
+        copyVecToArray(lowerBounds, numberColumns, columnLower);
+        copyVecToArray(upperBounds, numberColumns, columnUpper);
+
+        // Set rows bounds
+        copyVecToArray(constraintLowerBounds, numberRows, rowLower);
+        copyVecToArray(constraintUpperBounds, numberRows, rowUpper);
+
+        // Set constraints matrix
+        for (int j = 0; j < numberColumns; j++) {
+            starts[j] = j * numberRows;
+            copyVecToArray(constraintMatrix.col(j), numberRows, elements + j * numberRows);
+        }
+
+        starts[numberColumns] = numberElements;
+
+        // assign to matrix
+        CoinPackedMatrix * matrix = new CoinPackedMatrix(true, 0.0, 0.0);
```

Comment:
Tady to `new` vypadá podezřele, ale neznám knihovnu, nedokážu posoudit jestli to je špatně

Reply:
O uvolnění se postará destruktor ClpSimplex, doplněna dokumnetace.